Connect to private HCP cluster using public NLB

nedoshi opened the pull request.

nedoshi committed:
add docs

nedoshi committed:
fix links and updated images

nedoshi committed:
Update _index.md

nedoshi committed:
add to main index

nedoshi committed:
Update _index.md

nedoshi committed:
updated grammar and title

nedoshi committed:
updated typo

michaelryanmcneill commented on content/_index.md:
```diff
@@ -18,6 +18,7 @@ description: "Step-by-step tutorials from Red Hat experts to help you get the mo
 * [Securely exposing an application on a private ROSA cluster with a Network Load Balancer](/experts/rosa/hcp-private-nlb/)
 * [Add an Ingress Controller to ROSA with a custom domain](/experts/rosa/ingress-controller)
 * [Configuring Microsoft Entra ID as an external authentication provider](/experts/rosa/entra-external-auth)
+* [Accessing a Private ROSA Hosted Control Plane (HCP) Cluster with an AWS Network Load Balancer](/experts/rosa/hcp-public-nlb)
```
I am not sure we want to recommend this to customers as a home page item. Instead; it might be better to hide this from the homepage and just send direct links.

nedoshi replied:
I have removed the link from main index page

michaelryanmcneill reviewed (changes requested):
LGTM, just want to hide from the homepage.

nedoshi committed:
removed link from main index page

nedoshi commented:
I have removed this from main index page.Please approve it

paulczar commented:
Thanks for removing the guide from the homepage as requested! Before merging, I need one clarification:

**Missing version information:** The guide doesn't specify which ROSA version(s) this was tested on. Please provide:

1. Add `validated_version` to the frontmatter (e.g., `validated_version: "4.15"`)
2. Optionally, update the Prerequisites section to mention the tested ROSA version

For example:
- Prerequisites line 15 could read: "You will need a Private ROSA HCP cluster (4.15+)"

This ensures readers know the tested baseline and helps us flag the guide if that version reaches EOL.

Once updated, I can approve and merge. Thanks!

nedoshi committed:
Update versions

Validated on ROSA HCP cluster version 4.20 and updated doc

nedoshi commented:
@paulczar Updated the document with version

nedoshi committed:
Add guide for accessing private ROSA HCP cluster via public NLB

Document internet-facing NLB access to a private ROSA HCP cluster API,
including VPCE IP discovery, VPCE security group rules, and Entra ID
kubeconfig validation. Validated on ROSA HCP 4.20.

Signed-off-by: Nerav Doshi <103272684+nedoshi@users.noreply.github.com>
